api_app/db/repositories/workspaces.py

```diff
@@ -124,7 +128,10 @@ def validate_address_space(self, address_space):
         return is_network_available(allocated_networks, address_space)
 
     def get_new_address_space(self, cidr_netmask: int = 24):
-        networks = [x.properties["address_space"] for x in self.get_workspaces()]
+        workspaces = self.get_workspaces()
+        networks = [[x.properties.get("address_space")] for x in workspaces]
+        networks = networks + [x.properties.get("address_spaces", []) for x in workspaces]
+        networks = [i for s in networks for i in s if i is not None]
```

Comment:
What does this line do?

Reply:
flattens the `address_spaces` property, joins with `address_space`. So have a list of used address spaces so can find a free address space.

Reply:
I might be mistaken but a list + a list in python is still a list. 
So why is this line required?

Reply:
@tamirkamara which line specifically? 133?

Reply:
I mean that it looks like this is enough
```python
        workspaces = self.get_workspaces()
        networks = [[x.properties.get("address_space")] for x in workspaces]
        networks = networks + [x.properties.get("address_spaces", []) for x in workspaces]
```

Reply:
It flattens the lists, without that line the list of existing address spaces has multiple dimensions.

As in its a list address_space and address_spaces, where each address_spaces entry is actually another list. It needs to be flattened to be passed to the function that finds an unused address space.